lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -193,7 +194,14 @@ Step GcodeUpload::step(string_view input, PutParams &putParams) {
     // it can be used inside UploadState as a function of UploadHooks.
     if (!filename_checked) {
         auto filename_error = check_filename(filename);
-        if (std::get<0>(filename_error) != Status::Ok)
+        if (std::get<0>(filename_error) == Status::Conflict and putParams.overwrite) {
+            if (wui_is_file_being_printed(putParams.filepath.data())) {
```

Comment:
This seems like a bit of an abuse. I mean, it should probably be the job of check_filename to handle the situation, not guessing by what HTTP error code it returned and then trying to „fix“ the situation.